### PR TITLE
media-gfx/rawtherapee: version bump to 5.6

### PR DESCRIPTION
**Pull request by vivo75**
Author: Francesco Riosa <vivo75@gmail.com>
Date:   Wed Apr 24 09:12:02 2019 +0200

    media-gfx/rawtherapee: version bump to 5.6

    also added dependancy on librsvg >= 2.40, see
    https://rawtherapee.com/blog/rawtherapee-5.6-released

    Closes: https://bugs.gentoo.org/684154

    Signed-off-by: Francesco Riosa <vivo75@gmail.com>

### CODE REVIEW

File: media-gfx/rawtherapee/rawtherapee-5.6.ebuild [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+		-DCREDITSDIR=/usr/share/${PN}
+		-DLICENCEDIR=/usr/share/${PN}
+		-DCACHE_NAME_SUFFIX=""
+		-DWITH_SYSTEM_KLT="off"
+	)
+	cmake-utils_src_configure
+}
+
+pkg_postinst() {
+	gnome2_icon_cache_update
```
Comment (juippis) [inline media-gfx/rawtherapee/rawtherapee-5.6.ebuild]: You should switch to `xdg_icon_cache_update` which allows you to bump EAPI from this ebuild to 7 by removing gnome2-utils from inherit line.

File: media-gfx/rawtherapee/rawtherapee-5.6.ebuild [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+	media-libs/tiff:0
+	sci-libs/fftw:3.0
+	sys-libs/zlib
+	virtual/jpeg:0
+	x11-libs/gtk+:3"
+DEPEND="${RDEPEND}
+	app-arch/xz-utils
+	dev-cpp/gtkmm:3.0
+	gnome-base/librsvg
+	virtual/pkgconfig"
```
Comment (juippis) [inline media-gfx/rawtherapee/rawtherapee-5.6.ebuild]: If you do bump EAPI to 7, this is then a BDEPEND dependency instead.

**Comment by vivo75** [comment]
Thanks @juippis EAPI bump worked first try with your suggestions

File: media-gfx/rawtherapee/rawtherapee-5.6.ebuild (hunk 2 above)
Comment (juippis) [reply media-gfx/rawtherapee/rawtherapee-5.6.ebuild]: That went a bit wrong :) 
```
RDEPEND="dev-libs/expat
	dev-libs/libsigc++:2
	media-libs/lcms:2
	media-libs/lensfun
	media-libs/libcanberra[gtk3]
	media-libs/libiptcdata
	media-libs/libpng:0
	media-libs/tiff:0
	sci-libs/fftw:3.0
	sys-libs/zlib
	virtual/jpeg:0
	x11-libs/gtk+:3"
DEPEND="${RDEPEND}
	app-arch/xz-utils
	dev-cpp/gtkmm:3.0
	gnome-base/librsvg"
BDEPEND="virtual/pkgconfig"
```
would be the correct form. I wasn't sure about xz-utils, if their binaries just get executed and no linking happens or no header files needed, then it's a BDEPEND too. 

BDEPEND can be a bit hard to understand, but https://dev.gentoo.org/~mgorny/articles/the-ultimate-guide-to-eapi-7.html#build-time-dependencies-split-into-depend-and-bdepend tries to explain it. It's important to note that BDEPEND is a split off from DEPEND, thus can "never" inherit contents from RDEPEND or DEPEND.

Comment (vivo75) [reply media-gfx/rawtherapee/rawtherapee-5.6.ebuild]: @juippis for some reason github does not send me emails for comments, please mention me or the risk is that I lose the comment forever
I've checked the xz-utils dep and it's not used at compile time (except from portage when unpacking) also a runtime it's only a dep of gtk not directly linked by rawtherapee. Has been removed
The change you suggest is now applied

File: media-gfx/rawtherapee/rawtherapee-5.6.ebuild [hunk 3]
```diff
@@ -0,0 +1,71 @@
+# Copyright 1999-2019 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+inherit cmake-utils flag-o-matic toolchain-funcs xdg-utils
+
+MY_P=${P/_rc/-rc}
```
Comment (a17r) [inline media-gfx/rawtherapee/rawtherapee-5.6.ebuild]: lgtm, without having test-run myself yet. this line would be intuitively above `inherit` though, by convention of majorit of ebuilds.

File: media-gfx/rawtherapee/rawtherapee-5.6.ebuild [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+DESCRIPTION="A powerful cross-platform raw image processing program"
+HOMEPAGE="http://www.rawtherapee.com/"
```
Comment (a17r) [inline media-gfx/rawtherapee/rawtherapee-5.6.ebuild]: I had already changed this to `https` on latest ebuild, is this based on an older ebuild somehow?

Comment (vivo75) [reply media-gfx/rawtherapee/rawtherapee-5.6.ebuild]: yes, I had changes reqested at the time we pushed rawtherapee 5.5 and it didn't came to mind to also compare the in-tree ebuild to the one I have in overlay. Note taken for 5.7

**Pull request closed without merging**